webpack/assets/javascripts/react_app/components/ForemanModal/subcomponents/ForemanModalHeader.test.js

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import { Modal } from 'patternfly-react';
+import ForemanModalHeader from './ForemanModalHeader';
+import * as ModalContext from '../ForemanModal.hooks'; // so enzyme test works
+import { testComponentSnapshotsWithFixtures } from '../../../common/testHelpers';
+
+const fixtures = {
+  'should render with default markup': {
+    title: 'foo',
+  },
+  'should render with supplied children': {
+    title: 'should not be in markup',
+    children: <h1>Modal Title</h1>,
+  },
+};
+
+const contextValues = {
+  title: 'modal title passed thru mock context :)',
+};
+
+jest
+  .spyOn(ModalContext, 'useModalContext')
+  .mockImplementation(() => contextValues);
+
+describe('ForemanModal.Header', () => {
+  describe('rendering', () => {
+    testComponentSnapshotsWithFixtures(ForemanModalHeader, fixtures);
+  });
+  describe('data flow', () => {
+    it('renders a <Modal.Title> and title prop when no children are given', () => {
+      const wrapper = shallow(<ForemanModalHeader />);
+      expect(wrapper.find(Modal.Title)).toHaveLength(1);
+      expect(
+        wrapper
+          .find(Modal.Title)
+          .dive()
+          .text()
+      ).toMatch(contextValues.title);
+    });
+    it('ignores title prop if children are given', () => {
+      const wrapper = shallow(
+        <ForemanModalHeader>
+          <div>custom title</div>
+        </ForemanModalHeader>
+      );
+      expect(wrapper.find(Modal.Title)).toHaveLength(0);
+    });
+    it('passes props to PF component using spread', () => {
+      const wrapper = shallow(<ForemanModalHeader myCustomProp="hi" />);
+      expect(wrapper.find(Modal.Header).prop('myCustomProp')).toEqual('hi');
+    });
+    it ('has a close button by default');
+    it ('has no close button if overridden via props');
```

Comment:
Should we remove this two? thanks

Reply:
@laviro I forgot to write those tests ;) I think I'll just finish them instead